public/app/features/plugins/extensions/utils.tsx

```diff
@@ -218,12 +226,96 @@ export function getReadOnlyProxy<T extends object>(obj: T): T {
   });
 }
 
+/**
+ * Returns a proxy that logs any attempted mutation to the original object.
+ *
+ * @param obj The object to observe
+ * @returns A new proxy object that logs any attempted mutation to the original object
+ */
+export function getMutationObserverProxy<T extends object>(obj: T): T {
+  if (!obj || typeof obj !== 'object' || isMutationObserverProxy(obj)) {
+    return obj;
+  }
+
+  const cache = new WeakMap();
+
+  return new Proxy(obj, {
+    deleteProperty(target, prop) {
+      console.warn(`Extensions: Attempted to delete object property "${String(prop)}"`, new Error().stack);
+      delete target[prop as keyof T];
+      return true;
+    },
+    defineProperty(target, prop, descriptor) {
+      console.warn(`Extensions: Attempted to define object property "${String(prop)}"`, new Error().stack);
+      Object.defineProperty(target, prop as keyof T, descriptor);
+      return true;
+    },
+    set(target, prop, newValue) {
+      console.warn(`Extensions: Attempted to mutate object property "${String(prop)}"`, new Error().stack);
+      target[prop as keyof T] = newValue;
+      return true;
```

Comment:
Not sure because I'm not used to it but should we use `logWarning` function instead for adding these logs with mutations? 
https://github.com/grafana/grafana/blob/main/packages/grafana-runtime/src/utils/logging.ts#L25

Reply:
That's a really good point, will fix 👍

Reply:
Can we include the plugin Id in those messages if you think that would be a good addition?

Reply:
Good point @hugohaggmark! I have started to use the [`ExtensionsLog` service](https://github.com/grafana/grafana/blob/main/public/app/features/plugins/extensions/logs/log.ts#L23) for logging, which has this information in its labels. Also sending logs to Faro, so we can track the errors.